content/plugins/kubernetes.md

```diff
@@ -32,20 +32,107 @@ See [releases](https://github.com/cloudify-incubator/cloudify-kubernetes-plugin/
 This example demonstrates how to add resource definition in your node template.
 
 ```yaml
-  nginx_pod:
-    type: cloudify.kubernetes.resources.Pod
+  my_application:
+    type: cloudify.kubernetes.resources.MultipleFileDefinedResources
     properties:
-      definition:
-        apiVersion: v1
-        metadata:
-          name: nginx
-        spec:
-          containers:
-          - name: nginx
-            image: nginx:1.7.9
-            ports:
-            - containerPort: 80
+      files:
+        - resource_path: resources/my_app_service.yaml
+        - resource_path: resources/my_app_pod.yaml
     relationships:
       - type: cloudify.kubernetes.relationships.managed_by_master
-        target: kubernetes_master
+        target: master
+
+  master:
+    type: cloudify.kubernetes.nodes.Master
+    properties:
+      configuration:
+        file_content: { get_input: kubernetes_configuration_file_content }
+```
+
+**Many more examples are available [here](https://github.com/cloudify-incubator/cloudify-kubernetes-plugin/tree/master/examples).**
+
+# Types
+
+## cloudify.kubernetes.nodes.Master
+This node represents an existing Kubernetes master.
+
+### Properties:
+  * `configuration`: A master config, one of the following:
+    1. Kubernetes config file contained by blueprint archive
+    2. Kubernetes config file previously uploaded into Cloudify Manager VM
+    3. Content of Kubernetes config file (YAML)
+    4. Kubernetes API set of properties
+
+## cloudify.kubernetes.resources.FileDefinedResource
+This is a Kubernetes resource, such as a pod, service, deployment, which is defined in a file.
+
+### Properties:
+  * `file`: The path to the file relative to the blueprint.
+  * `target_path`: The path the file should be downloaded to. (Don't use this.)
+  * `template_variables` If the file has Jinja template variable references, you may provide the variables as a dictionary here.
+
+## cloudify.kubernetes.resources.MultipleFileDefinedResources
+
+Provide a list of files containing Kubernetes resources.
+
+### Properties:
+
+  * `files` a list of resource_path's.
+
+
+## cloudify.kubernetes.resources.BlueprintDefinedResource
+This is the root type of all Kubernetes resource, such as a pod, service, deployment, which is defined in the blueprint inline.
+
+### Properties:
+  * `definition` The resource definition. (The content of a Kubernetes template file.) This may be a string or JSON or YAML.
+  * `options` Kubernetes API mappings, such as ```{ 'namespace': 'default' }```.
+
+### Derived resource types:
+
+  Check the [plugin.yaml](https://github.com/cloudify-incubator/cloudify-kubernetes-plugin/blob/master/plugin.yaml) for latest additions.
```

Comment:
Maybe we should specify which version of the plugin these components are applicable for?